test/unit/test-resources.js

```diff
@@ -1442,6 +1482,22 @@ describes.fakeWin('Resources.add/upgrade/remove', {amp: true}, (env) => {
     }
   );
 
+  it('should observe element after adding it', () => {
+    // Enables the 'intersect-resources' experiment.
+    const observer = (resources.intersectionObserver_ = {
+      observe: env.sandbox.spy(),
+    });
+    // Avoid creating a new Resource, which is tricky to spy on.
+    env.sandbox.stub(child1, 'reconstructWhenReparented').returns(false);
+    env.sandbox.stub(resource1, 'getState').returns(ResourceState.NOT_LAID_OUT);
```

Comment:
what does making `getState` return `NOT_LAID_OUT` buy us? My impression is that requestMeasure shouldn't be called either way, as well as that observe should be called either way.

Reply:
It's needed to enter this conditional: https://github.com/ampproject/amphtml/blob/7c8b09bd2a390921ba503ed89fd6075134672fb4/src/service/resources-impl.js#L406-L421

Reply:
Anything other than `NOT_BUILT` would have been fine though right? I'm not saying you should have picked a different one since `NOT_LAID_OUT` is next, just making sure I understand properly

Reply:
Right! I think it's possible for an element to be reparented after layout completes.